holopy/core/tests/test_io.py

```diff
@@ -230,6 +230,22 @@ def __init__(self, a, b, c=3):
         # only a & c appear because b is None, d is not in __init__ signature
         assert yaml.dump(instantiated) == '!S\na: 1\nc: 3\n'
 
+    @attr("fast")
+    def test_custom_treatment_of_ndarrays(self):
+        zero_length_array = np.array(3)
+        one_length_array = np.array([3])
+        longer_array = 3 * np.ones(10, dtype=int)
```

Comment:
it might be worth splitting this up into 4 separate tests, that way if some change down the line in xarray or yaml does break this test, it will be more obvious which sub-cases need to be fixed.

Reply:
done